### PR TITLE
Add JSON support

### PR DESCRIPTION
**Pull request by jayvdb**
Closes https://github.com/Electron100/butane/issues/27

Tests are passing.

~~Depends on https://github.com/Electron100/butane/pull/37 to clear most of the lint errors.  Then I can use clippy to tidy this PR up a bit.~~

### CODE REVIEW

File: butane_core/src/sqlval.rs [hunk 1]
```diff
@@ -253,6 +265,8 @@ impl<'a> From<&'a SqlVal> for SqlValRef<'a> {
             Real(v) => SqlValRef::Real(*v),
             Text(v) => SqlValRef::Text(v.as_ref()),
             Blob(v) => SqlValRef::Blob(v.as_ref()),
+            #[cfg(feature = "json")]
+            Json(v) => SqlValRef::Json(v.clone()), // should be as_ref? doesnt compile. *v also doesnt compile
```
Comment (jayvdb) [inline butane_core/src/sqlval.rs:269]: This needs to be fixed.

File: butane/Cargo.toml [hunk 2]
```diff
@@ -12,7 +12,8 @@ repository = "https://github.com/Electron100/butane"
 documentation = "https://docs.rs/butane/"
 
 [features]
-default = ["datetime", "uuid"]
+default = ["datetime", "json", "uuid"]
```
Comment (jayvdb) [inline butane/Cargo.toml:15]: I've enabled it by default as serde_json is already a dependency of butane, and this gives butane the ability store almost any data structure, even if it isnt in the format that they want it stored it.

File: butane/tests/json.rs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+            "age": 43,
+            "phones": [
+                "+44 1234567",
+                "+44 2345678"
+            ]
+        }"#;
+
+    foo.val = serde_json::from_str(data).unwrap();
+    foo.save(&conn).unwrap();
+    println!("{}", Value::default());
```
Comment (jayvdb) [inline butane/tests/json.rs:60]: need to remove this before next push

File: butane/Cargo.toml (hunk 2 above)
Comment (Electron100) [reply butane/Cargo.toml:15]: Fair enough

**Comment by jayvdb** [comment]
@Electron100 , Thanks for merging #37 - this PR is now green and ready for review.

**Review by Electron100 — APPROVED** [review]

**Comment by Electron100** [comment]
Thanks for doing this!

**Pull request merged**